tests/test_internal.py

```diff
@@ -239,7 +239,7 @@ def test_two_header_detection(self):
             list(fq)
 
     def test_second_header_not_equal(self):
-        fastq = BytesIO(b'@r1\nACG\n+xy\n')
+        fastq = BytesIO(b'@r1\nACG\n+xy\nXXX\n')
```

Comment:
Can you please duplicate the test and test it with and without the trailing `XXX`?

Reply:
I first check if the quality length in order, because I do a shortcut there. But that is not really necessary. I can revert the change to this test if I do the checks in the correct order again. Will do so after the entire review.

Reply:
I Looked a bit at it. Removing the XXX quality line will give a Premature End Of File error. Which isn't wrong in this case. 
Due to the code reorganization a check for a comple fastq record is done first before any other checking or parsing is done. This is optimal, because reaching the end of the buffer happens quite often, and we want to refill the buffer ASAP instead of checking a FASTQ that is not complete anyway.

Reply:
I don’t quite understand. I just want the existing test to work as before. If you want, you can of course add another test, but don’t change this one.